MKPC pulls

Gapva opened the pull request:
Fixed very few grammatical errors within the course builder and in other places

tmalahie committed:
Small fixes in base setup

Gapva committed:
Merge branch 'development' of https://github.com/tmalahie/mkpc into development

Gapva committed:
Merge branch 'development' of https://github.com/tmalahie/mkpc into development

Gapva committed:
Fixed grammatical errors

Gapva committed:
Fixed grammatical errors

"You use" --> "You're using"

Gapva committed:
Fixed grammatical errors

"Face off" --> "Face-off"
"cpu" --> "CPU"

Gapva committed:
More Framerates + Typo Fix

Added support for 90hz (mobile) devices, 144hz and 240hz monitors. fixed a typo where the "r" in "frame" was missing in "interframe"

Gapva committed:
Button animations + More options + Text fixes

Added button hover animations and some additional design to them as well; added Tiny and Massive (translated both languages, checked with Fways), fixed very few capitalization errors

Gapva committed:
Added smooth pause dim effect

Edited pause hotkey script and pause function (incase of touchscreen) to make this work :)

tmalahie commented on scripts/mk.js:
```diff
@@ -3870,6 +3872,7 @@ var fLineScale = 0;
 // setup main container
 var oContainers = [document.createElement("div")];
 oContainers[0].className = "game-container";
+oContainers[0].id = "game-container"
```
You don't need this id, `oContainers[0]` is already the same as `document.getElementById("game-container")`

Gapva replied:
thought so, just felt like we needed an id but that also works :)

tmalahie commented on scripts/mk.js:
```diff
@@ -4039,6 +4042,7 @@ function interruptGame() {
 function reprendre(debug) {
 	if(pause) {
 		pause = false;
+		document.getElementById("game-container").style.filter = "brightness(100%)";
```
```suggestion
		oContainers[0].style.filter = "brightness(100%)";
```

tmalahie commented on scripts/mk.js:
```diff
@@ -3346,6 +3347,7 @@ function startGame() {
 											var resumeButton = document.getElementById("reprendre");
 											if (resumeButton)
 												resumeButton.focus();
+												document.getElementById("game-container").style.filter = "brightness(25%)";
```
```suggestion
									oContainers[0].style.filter = "brightness(25%)";
```

tmalahie commented on scripts/mk.js:
```diff
@@ -3156,6 +3156,7 @@ function startGame() {
 									btnDir = 1;
 									break;
 								}
+								document.getElementById("game-container").style.filter = "brightness(100%)";
```
```suggestion
								oContainers[0].style.filter = "brightness(100%)";
```

tmalahie replied:
Why did you add this line btw? It makes the effect disappear when I hit a key while in pause, not sure why we would want this :thinking:

Gapva replied:
oh, oops. my bad 😅 I was about to sleep and never really considered stuff like this

tmalahie commented on styles/mariokart.css:
```diff
@@ -14,6 +14,20 @@
+  div div input:hover {
+	transform: rotate(-5deg) scale(85%);
+	filter: brightness(200%);
+}
```
Targetting all inputs inside div's divs seem quite impactful, have you tested it doesn't have side-effects?

tmalahie replied:
I'd suggest limiting to `button`s and `input[type="button"]` (I already do it line 503), otherwhise it also affects text inputs and radio inputs, which is kinda strange

tmalahie commented on scripts/mk.va4.js:
```diff
@@ -1 +1 @@
-var pause,chatting=!1,aPlayers=new Array,aPlaces=new Array,aScores=new Array,aTeams=new Array,aPseudos=new Array,aControllers=new Array,fInfos,formulaire,baseCp,baseCp0,pUnlockMap,customDecorData={},nBasePersos,customPersos,selectedDifficulty,updateCtnFullScreen,isFirstLoad=!0,selectedCc=localStorage.getItem("cc")||"150",selectedNbTeams=localStorage.getItem("nbTeams")||"2",selectedFriendlyFire=!! ...
```
You don't need to change those files, I usually auto-generate them when I make a release

tmalahie reviewed (commented):
Those are great suggestions, but I think we can improve it before going to production

Gapva committed:
Update scripts/mk.js

Co-authored-by: Malahieude Timothé <t.malahieude@gmail.com>

Gapva committed:
Update scripts/mk.js

Co-authored-by: Malahieude Timothé <t.malahieude@gmail.com>

Gapva committed:
Update scripts/mk.js

Co-authored-by: Malahieude Timothé <t.malahieude@gmail.com>

Gapva committed:
Minimize ambiguity in pause menu

"CHANGE RACE" --> "CHANGE TRACK";
There are no issues with the French translation, so I did not tamper with it.

Gapva committed:
Merge remote-tracking branch 'upstream/development' into development
